ml/misc/linked_list.py

```diff
@@ -0,0 +1,260 @@
+import copy
+
+
+# Build a LinkedList class and implement reverse function
+class LinkedList:
+
+    def __init__(self, init_list=[]):
+        self.head = None
+        self.length = 0
+        self.from_list(init_list)
+        pass
+
+    def from_list(self, init_list):
+        if not isinstance(init_list, list):
+            init_list = []
+        current = None
+        for v in init_list:
+            item = copy.deepcopy(v)
+            next_item = {
+                'data': item,
+                'next': None,
+            }
+            if current is None:
+                self.head = next_item
+            else:
+                current['next'] = next_item
+            current = next_item
+        self.length = len(init_list)
+        # self.print()
+
+    def add(self, input_item, position=0):
+        """
+        add input_item to certain position, if the input position is larger than the length of original head, will
+        place the input item to the end, and put to the begin if position <= 0
+        @param input_item: item to be added
+        @param position: position to be added
+        @return:
+        """
+        if self.head is None:
+            self.head = {
+                'data': input_item,
+                'next': None,
+            }
+            return
+        if not isinstance(position, int):
+            raise TypeError
```

Comment:
Optionally to return a non-None item (`new_item`) if `add` succeeded; otherwise, return `None`.